v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java

```diff
@@ -0,0 +1,249 @@
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.conditions.ChainedConditionCheck;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.beam.it.gcp.spanner.SpannerResourceManager;
+import org.apache.beam.it.gcp.spanner.conditions.SpannerRowsCheck;
+import org.apache.beam.it.gcp.spanner.matchers.SpannerAsserts;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A Simple Integration test for {@link DataStreamToSpanner} Flex template. */
+@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})
+@TemplateIntegrationTest(DataStreamToSpanner.class)
+@RunWith(JUnit4.class)
+public class DataStreamToSpannerSimpleIT extends DataStreamToSpannerITBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DataStreamToSpannerSimpleIT.class);
+
+  private static final String TABLE1 = "Users";
+  private static final String TABLE2 = "Movie";
+
+  private static final String SESSION_FILE_RESOURCE =
+      "DataStreamToSpannerSimpleIT/mysql-session.json";
+
+  private static final String SPANNER_DDL_RESOURCE =
+      "DataStreamToSpannerSimpleIT/spanner-schema.sql";
+
+  private static HashSet<DataStreamToSpannerSimpleIT> testInstances = new HashSet<>();
+  private static PipelineLauncher.LaunchInfo jobInfo;
+
+  public static PubsubResourceManager pubsubResourceManager;
+  public static SpannerResourceManager spannerResourceManager;
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException {
+    // Prevent cleaning up of dataflow job after a test method is executed.
+    skipBaseCleanup = true;
+    testInstances.add(this);
+    if (jobInfo == null) {
+      spannerResourceManager = setUpSpannerResourceManager();
+      pubsubResourceManager = setUpPubSubResourceManager();
+      jobInfo =
+          launchDataflowJob(
+              getClass().getSimpleName(),
+              SESSION_FILE_RESOURCE,
+              SPANNER_DDL_RESOURCE,
+              spannerResourceManager,
+              pubsubResourceManager,
+              new HashMap<>() {
+                {
+                  put("inputFileFormat", "json");
+                }
+              });
+    }
+  }
```

Comment:
What is the reasoning behind preventing resource cleanup after each test? Tests are run in parallel and we typically use non-static resource managers that spin up and clean up before and after each test.

Reply:
This test is written such that one single Dataflow job is run for all the test methods in the class. 
1. Time taken to launching a dataflow job with flex template and that job processing the events takes ~8 minutes of time. So, having one DF job for each test method is costly. A dataflow job once started can be used to test multiple test scenarios.
2. We do not want to club multiple test scenarios in the same test method. The approach in this PR keeps test methods small and modular while also shortening the time taken for setup.

That's why the resource managers are static and I am preventing resource cleanup after each test method. The resource cleanup happens in the AfterClass annotation which would be executed after all the test methods are done executing.